cocotb/decorators.py

```diff
@@ -602,3 +293,22 @@ def __call__(self, *args, **kwargs):
         inst = self._func(*args, **kwargs)
         coro = RunningTest(inst, self)
         return coro
+
+
+def __getattr__(attr: str) -> typing.Any:
```

Comment:
I think this should be:
```suggestion
def __getattribute__(attr: str) -> typing.Any:
```
The imported classes will be found and `__getattr__()` won't be called.
https://docs.python.org/3.6/reference/datamodel.html?highlight=__getattr__#object.__getattribute__

Reply:
Unfortunately modules do not support `__getattribute__`, but good catch. I'll have to think of something else.

Reply:
Instead I imported the names privately and in the `__getattr__` adjusted the `attr` argument to point to the private name.